Source/Ultraviolet/Shared/Graphics/VertexPositionNormalTexture.cs

```diff
@@ -0,0 +1,53 @@
+﻿namespace Ultraviolet.Graphics
+{
+    /// <summary>
+    /// Represents a vertex containing 3D position, normal, and texture coordinate data.
+    /// </summary>
+    public struct VertexPositionNormalTexture : IVertexType
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="VertexPositionNormalTexture"/> structure.
+        /// </summary>
+        /// <param name="position">The vertex position.</param>
+        /// <param name="normal">The vertex normal.</param>
+        /// <param name="textureCoordinate">The texture coordinate.</param>
+        public VertexPositionNormalTexture(Vector3 position, Vector3 normal, Vector2 textureCoordinate)
+        {
+            this.Position          = position;
+            this.Normal            = normal;
+            this.TextureCoordinate = textureCoordinate;
+        }
+
+        /// <summary>
+        /// Gets the vertex declaration.
+        /// </summary>
+        VertexDeclaration IVertexType.VertexDeclaration
+        {
+            get { return VertexDeclaration; }
+        }
+
+        /// <summary>
+        /// The vertex declaration.
+        /// </summary>
+        public static readonly VertexDeclaration VertexDeclaration = new VertexDeclaration(new[] {
+            new VertexElement(0, VertexFormat.Vector3, VertexUsage.Position, 0),
+            new VertexElement(sizeof(Single) * 3, VertexFormat.Vector3, VertexUsage.Normal, 0),
+            new VertexElement(sizeof(Single) * 3 + sizeof(Byte) * 4, VertexFormat.Vector2, VertexUsage.TextureCoordinate, 0),
```

Comment:
This line looks like a copy-paste error. The texture coordinate is preceded by six `Single` values, not three `Single` values and four `Byte` values.